sshd-core/src/test/java/org/apache/sshd/common/config/keys/OpenSSHKeyUtilsTest.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.sshd.common.config.keys;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.security.NoSuchProviderException;
+import java.security.PublicKey;
+
+import org.apache.sshd.common.config.keys.KeyUtilsTest;
+import org.apache.sshd.common.digest.BuiltinDigests;
+import org.apache.sshd.common.digest.Digest;
+import org.apache.sshd.common.digest.DigestFactory;
+import org.apache.sshd.common.util.Pair;
+import org.apache.sshd.util.test.BaseTestSupport;
+import org.junit.BeforeClass;
+import org.junit.FixMethodOrder;
+import org.junit.Test;
+import org.junit.runners.MethodSorters;
+
+/**
+ * @author <a href="mailto:dev@mina.apache.org">Apache MINA SSHD Project</a>
+ */
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class OpenSSHKeyUtilsTest extends BaseTestSupport {
+
+    private static class KeyEntry {
+        public final String keyString;
+        public final Pair<DigestFactory, String>[] hashes;
+        public PublicKey key;
+        public KeyEntry(String keyString, Pair<DigestFactory, String>... hashes) {
+            this.keyString = keyString;
+            this.hashes = hashes;
+        }
+    }
```

Comment:
I prefer:
```java
private static class KeyEntry {
        public final String keyString;
        public final List<Pair<DigestFactory, String>> hashes;
        public PublicKey key;
        public KeyEntry(String keyString, Pair<DigestFactory, String>... hashes) {
            this.keyString = keyString;
            this.hashes = Collections.unmodifiableList(Arrays.asList(hashes));
        }
    }
```
This ensure that the test code does not inadvertently change the tested data